SDL2_classic: 2.32.2 -> 2.32.4

trofi opened the pull request:
Changes: https://github.com/libsdl-org/SDL/releases/tag/release-2.32.4


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

trofi committed:
SDL2_classic: 2.32.2 -> 2.32.4

Changes: https://github.com/libsdl-org/SDL/releases/tag/release-2.32.4

LordGrimmauld commented on pkgs/by-name/sd/SDL2_classic/package.nix:
```diff
@@ -86,19 +85,6 @@ stdenv.mkDerivation (finalAttrs: {
-    # On a normal distro this is enough for includes from all SDL2* packages to work,
-    # but on NixOS they're spread across different paths.
-    # This patch + the setup-hook will ensure that `sdl2-config --cflags` works correctly.
-    ./find-headers.patch
```
you probably only intended to drop the commit that is now included in the release, not the header patch? Otherwise, please also remove the header patch file.

trofi replied:
Dropped by accident. Restored.

LordGrimmauld reviewed (commented):
This doesn't need to go to staging anymore. Also, CI complains about your formatting, please fix.

trofi commented:
> This doesn't need to go to staging anymore.

Good point. Switched to `master`.

> Also, CI complains about your formatting, please fix.

Looking at the failure I thing the failure is unrelated to `SDL2_classic` changes but is related to something called `playwright`:

```diff
--- a/pkgs/development/web/playwright/webkit.nix
+++ b/pkgs/development/web/playwright/webkit.nix
@@ -80,7 +80,7 @@ let
         hash = "sha256-ME/mkaHhFeHajTbc7zhg9vtf/8XgkgSRu9I/mlQXnds=";
       };
       postPatch = "";
-      patches = [];
+      patches = [ ];
     }
   );
```

Rebased against a newer merge base hoping it was fixed since.

LordGrimmauld commented:
Commit message/PR title is inaccurate, old version was 2.32.**2**

LordGrimmauld reviewed (approved):
I checked upstream diff, looks fine.
Running nixpkgs-review rn, but apart from the commit message inaccuracy this lgtm.

trofi commented:
Fixed commit/PR descriptions to mention `2.32.2` source version.

LordGrimmauld reviewed (approved):
Most of the failing python things are broken on master until https://github.com/NixOS/nixpkgs/pull/395688

This here looks fine to me, thanks
